src/typescript/frontend/src/pages/launch-emojicoin/index.tsx

```diff
@@ -50,11 +59,42 @@ const LaunchEmojicoinPage: React.FC = () => {
   );
 
   const onEmojiClickHandler = async (emoji: EmojiClickData) => {
-    await setFieldValue("emoji", emoji.emoji);
-    setEmoji(emoji);
-    console.log("emoji", emoji);
+    await setFieldValue("emoji", values.emoji + emoji.emoji);
+    await setFieldValue("emojiList", [...values.emojiList, emoji]);
+  };
+
+  const inputProhibition = async (event: KeyboardEvent) => {
+    if (event.key !== "Backspace") {
+      event.preventDefault();
+    } else {
+      const newEmojiValue = values.emoji.slice(0, -1);
+      const newEmojiListValue = values.emojiList.slice(0, -1);
+      await setFieldValue("emoji", newEmojiValue);
+      await setFieldValue("emojiList", newEmojiListValue);
+    }
   };
 
+  const cursorPositioning = () => {
+    if (input) {
+      input.focus();
+      input.selectionStart = input.value.length;
+    }
+  };
+
+  useEffect(() => {
+    if (input) {
+      input.addEventListener("keydown", inputProhibition);
+      input.addEventListener("click", cursorPositioning);
+    }
+
+    return () => {
+      if (input) {
+        input.removeEventListener("keydown", inputProhibition);
+        input.removeEventListener("click", cursorPositioning);
+      }
+    };
+  }, [input, values]);
```

Comment:
I think you can actually move all the `input` stuff, aka `inputProhibition`, `cursorPositioning`, `eventListeners` etc inside of the `useEffect`

If you refactor to an `inputRef` as well you can put `const input = inputRef.current` in there, too